src/Node.tsx

```diff
@@ -50,13 +48,13 @@ export const Node = <T extends unknown>(props: Props): ReactElement | null => {
     if (treeContext.dragPreviewRender) {
       preview(getEmptyImage(), { captureDraggingState: true });
     }
-  }, []);
+  }, [preview, treeContext.dragPreviewRender]);
 
   useDragControl(ref);
 
   const handleToggle = useCallback(() => {
     treeContext.onToggle(item.id);
-  }, [openIds]);
+  }, [item.id, treeContext]);
```

Comment:
The `useCallback` is used here to memoize the `handleToggle`, but this memoization is meaningless, so you can remove the `useCallback` as shown below.

```
const handleToggle = () => treeContext.onToggle(item.id);
```